R/extract.R

```diff
@@ -1842,21 +2037,92 @@ extract.glm <- function(model, include.aic = TRUE, include.bic = TRUE,
   )
   return(tr)
 }
-
+#' \code{\link{extract}} method for \code{glm} objects
+#'
+#' \code{\link{extract}} method for \code{glm} objects. These objects are
+#' created by the \code{\link[stats]{glm}} function in the \pkg{stats} package.
+#'
+#' @param model A model object.
+#' @param include.aic Report AIC in the GOF block?
+#' @param include.bic Report BIC in the GOF block?
+#' @param include.loglik Report the log likelihood in the GOF block?
+#' @param include.deviance Report the deviance in the GOF block?
+#' @param include.nobs Report the number of observations in the GOF block?
+#' @param ... Additional arguments for the \code{\link[base]{summary}} function
+#'   for \code{glm} objects.
+#' @return A \linkS4class{texreg} object.
+#'
+#' @method extract glm
+#' @aliases extract.glm
+#' @family extract
+#' @seealso \link{extract}
+#' @author Philip Leifeld
+#'
+#' @importFrom stats nobs
+#' @export
 setMethod("extract", signature = className("glm", "stats"),
     definition = extract.glm)
 
 extract.brglm <- extract.glm
+
+#' \code{\link{extract}} method for \code{brglm} objects
+#'
+#' \code{\link{extract}} method for \code{brglm} objects. These objects are
+#' created by the \code{\link[brglm]{brglm}} function in the \pkg{brglm}
+#' package.
+#'
+#' @inheritParams extract,glm-method
+#' @return A \linkS4class{texreg} object.
+#'
+#' @method extract brglm
+#' @aliases extract.brglm
+#' @family extract
+#' @seealso \link{extract}
+#' @author Philip Leifeld
+#'
+#' @export
 setMethod("extract", signature = className("brglm", "brglm"),
     definition = extract.glm)
 
 extract.negbin <- extract.glm
+
+#' \code{\link{extract}} method for \code{negbin} objects
+#'
+#' \code{\link{extract}} method for \code{negbin} objects. These objects are
+#' created by the \code{\link[MASS]{negbin}} function in the \pkg{negbin}
+#' package.
+#'
+#' @inheritParams extract,glm-method
+#' @return A \linkS4class{texreg} object.
+#'
+#' @method extract negbin
+#' @aliases extract.negbin
+#' @family extract
+#' @seealso \link{extract}
+#' @author Philip Leifeld
+#'
+#' @export
 setMethod("extract", signature = className("negbin", "MASS"),
     definition = extract.negbin)
 
-
-#extension for speedglm objects
 extract.speedglm <- extract.glm
+
+#' \code{\link{extract}} method for \code{speedglm} objects
+#'
+#' \code{\link{extract}} method for \code{speedglm} objects. These objects are
+#' created by the \code{\link[speedglm]{speedglm}} function in the \pkg{speedglm}
+#' package.
+#'
+#' @inheritParams extract,glm-method
+#' @return A \linkS4class{texreg} object.
+#'
+#' @method extract speedglm
+#' @aliases extract.speedglm
+#' @family extract
+#' @seealso \link{extract}
+#' @author Philip Leifeld
+#'
```

Comment:
I guess it might make sense to document `speedglm` together on one Rd page with `speedlm`. See how I did this with the `lme4` methods.

Reply:
here same question as above. They have different classes. Is it doable? Thanks

Reply:
Yes, it's like in the `lme4` case, where I documented the classes `lmerMod', 'glmerMod', and 'nlmerMod' on a single page. Look it up...